snapcraft/internal/project_loader.py

```diff
@@ -131,6 +132,15 @@ def __init__(self, project_options=None):
         snapcraft_yaml = self._expand_filesets(snapcraft_yaml)
         self.data = self._expand_env(snapcraft_yaml)
 
+        # Prevent multiple parts from having duplicate alias names
+        aliases = []
+        for app in self.data.get('apps', []):
+            app = self.data['apps'][app]
+            for alias in app.get('aliases', []):
```

Comment:
maybe something like this might be easier
```ipython
In [1]: l = ['1', '1', '2', '3']

In [2]: m = set(l)

In [3]: len(m)
Out[3]: 3

In [4]: len(l)
Out[4]: 4

In [5]: len(m) != len(l)
Out[5]: True
```